Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,326 @@
+    }
+    if ($IntraOrgCon.enabled -Like "False") {
+        hostOutputIntraOrgConNotEnabled
+    }
+    # Free busy Lookup methods
+    PrintDynamicWidthLine
+    $OrgRel = Get-OrganizationRelationship | Where-Object { ($_.DomainNames -like $ExchangeOnlineDomain) } | Select-Object Enabled, Identity, DomainNames, FreeBusy*, TarGet*
+    $EDiscoveryEndpoint = Get-IntraOrganizationConfiguration -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Select-Object OnPremiseDiscoveryEndpoint
+    $SPDomainsOnprem = Get-SharingPolicy -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Format-List Domains
+    $SPOnprem = Get-SharingPolicy  -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Select-Object *
+
+    if ($Org -contains 'ExchangeOnPremise' -or -not $Org) {
+        #region DAuth Checks
+        if ($Auth -like "DAuth" -OR -not $Auth -or $Auth -like "All") {
+            Write-Host $TestingDAuthConfiguration
+            OrgRelCheck -OrgRelParameter $OrgRel
+            PrintDynamicWidthLine
+            FedInfoCheck
+            FedTrustCheck
+            AutoDVirtualDCheck
+            PrintDynamicWidthLine
+            EWSVirtualDirectoryCheck
+            AvailabilityAddressSpaceCheck
+            TestFedTrust
+            TestOrgRel
+        }
+        #endregion
+        #region OAuth Check
+        if ($Auth -like "OAuth" -or -not $Auth -or $Auth -like "All") {
+            Write-Host $TestingOAuthConfiguration
+            # PrintDynamicWidthLine
+            IntraOrgConCheck
+            PrintDynamicWidthLine
+            AuthServerCheck
+            PrintDynamicWidthLine
+            PartnerApplicationCheck
+            PrintDynamicWidthLine
+            ApplicationAccountCheck
+            PrintDynamicWidthLine
+            ManagementRoleAssignmentCheck
+            PrintDynamicWidthLine
+            AuthConfigCheck
+            PrintDynamicWidthLine
+            CurrentCertificateThumbprintCheck
+            PrintDynamicWidthLine
+            AutoDVirtualDCheckOAuth
+            $AutoDiscoveryVirtualDirectoryOAuth
+            PrintDynamicWidthLine
+            EWSVirtualDirectoryCheckOAuth
+            PrintDynamicWidthLine
+            AvailabilityAddressSpaceCheckOAuth
+            PrintDynamicWidthLine
+            OAuthConnectivityCheck
+            PrintDynamicWidthLine
+        }
+        #endregion
+    }
+    # EXO Part
+    if ($Org -contains 'ExchangeOnline' -OR -not $Org) {
+        #region ConnectExo
+
+        Write-Host -ForegroundColor Green $CollectingExoAvailabilityInformation
+        # Check if the ExchangeOnlineManagement module is already installed
+        if (-not (Get-Module -Name ExchangeOnlineManagement -ListAvailable)) {
+            # If not installed, then install the module
+            Write-Host -ForegroundColor Yellow $ExchangeOnlinePowershellModuleMessage
+            Install-Module -Name ExchangeOnlineManagement -Force
+            PrintDynamicWidthLine
```

Comment:
Let's use `#Requires` for the module check

Reply:
If the user wishes to check only On Premises settings, Exchange Online Management module is not needed. This is very common on DAuth Free busy. Should I implement Requires in this case?